metaflow/plugins/kfp/tests/flows/foreach_linear_foreach.py

```diff
@@ -34,15 +37,51 @@ def linear_4(self):
 
     @step
     def foreach_join_z(self, inputs):
+        self.merge_artifacts(inputs)
         self.next(self.foreach_join_start)
 
     @step
     def foreach_join_start(self, inputs):
+        self.merge_artifacts(inputs)
         self.next(self.end)
 
     @step
     def end(self):
-        pass
+        print("Testing logging for foreach")
+        logging_step: Step = Step(f"{current.flow_name}/{current.run_id}/linear_2")
+        allowed_logs = [
+            self.log_testing_msg.format(data=data_input)
+            for data_input in self.x
+        ]
+        logs = []
+        print("Expect one of the following logs from each step: ", allowed_logs)
+        for task in logging_step.tasks():
+            log = task.stdout
+            print(f"=== Start: stdout from step index {task.index}===")
+            print(log)
+            print(f"=== End: stdout from step index {task.index}===")
+            assert any(log_candidate in log for log_candidate in allowed_logs)
```

Comment:
This line doesn't validate that the log is correct, just that it is one of the `allowed_logs`.  
What if every task log was `Testing logging. Data: a`?  This could happen if the s3 paths are incorrect and overwrite each other.